testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/security/authentication/BasicAuthMechanismServerSetupTask.java

```diff
@@ -0,0 +1,238 @@
+        writeAttrOp4.get(ClientConstants.VALUE).add("enabled", true);
+        writeAttrOp4.get(ClientConstants.VALUE).add("sasl-authentication-factory", getSecurityDomainName());
+        operations.add(writeAttrOp4);
+        // core-service=management/management-interface=http-interface:write-attribute(name=http-authentication-factory,value=management-http-authentication)
+        ModelNode writeAttrOp5 = createOpNode("core-service=management/management-interface=http-interface", WRITE_ATTRIBUTE_OPERATION);
+        writeAttrOp5.get(ClientConstants.NAME).set("http-authentication-factory");
+        writeAttrOp5.get(ClientConstants.VALUE).set(getSecurityDomainName());
+        operations.add(writeAttrOp5);
+        // core-service=management/management-interface=http-interface:undefine-attribute(name=security-realm)
+        ModelNode undefineAttrOp3 = createOpNode("core-service=management/management-interface=http-interface", UNDEFINE_ATTRIBUTE_OPERATION);
+        undefineAttrOp3.get(ClientConstants.NAME).set("security-realm");
+        operations.add(undefineAttrOp3);
+        // core-service=management/security-realm=ApplicationRealm/authentication=local:remove
+        ModelNode removeOp2 = createOpNode("core-service=management/security-realm=ApplicationRealm/authentication=local", REMOVE);
+        operations.add(removeOp2);
+        // core-service=management/security-realm=ApplicationRealm/authentication=properties:remove
+        ModelNode removeOp3 = createOpNode("core-service=management/security-realm=ApplicationRealm/authentication=properties", REMOVE);
+        operations.add(removeOp3);
+        // core-service=management/security-realm=ApplicationRealm/authorization=properties:remove
+        ModelNode removeOp4 = createOpNode("core-service=management/security-realm=ApplicationRealm/authorization=properties", REMOVE);
+        operations.add(removeOp4);
+
+        ModelNode updateOp = Operations.createCompositeOperation(operations);
+        updateOp.get(OPERATION_HEADERS, ROLLBACK_ON_RUNTIME_FAILURE).set(false);
+        updateOp.get(OPERATION_HEADERS, ALLOW_RESOURCE_SERVICE_RESTART).set(true);
+        CoreUtils.applyUpdate(updateOp, managementClient.getControllerClient());
+
+        ModelNode removeSecurityOp = new ModelNode();
+        removeSecurityOp.get(OP).set(REMOVE);
+        removeSecurityOp.get(OP_ADDR).add(SUBSYSTEM, "security");
+        CoreUtils.applyUpdate(removeSecurityOp, managementClient.getControllerClient());
+    }
+
+    /**
+     * Returns the server configuration to the initial state
+     *
+     * @param managementClient
+     * @param containerId
+     * @throws Exception
+     */
+    @Override
+    public void tearDown(ManagementClient managementClient, String containerId) throws Exception {
+        // ˇˇˇ Add security subsystem ˇˇˇ
+        List<ModelNode> operations = new ArrayList<>();
+        ModelNode addOp1 = createOpNode("subsystem=security", ADD);
```

Comment:
This really needs https://github.com/wildfly/wildfly/pull/11260 to be merged first, as this will allow server snapshots to be used to restore the server side state. This PR triggered an unrelated bug which caused it to be held up, but it should be possible to get it in soon.

Reply:
Can we merge this PR now, https://github.com/wildfly/wildfly/pull/11260 seems to be merge already. @stuartwdouglas  mentioned some bug, I dont know the state of this and if it is important for this PR to be merged. Do you have some information? Thanks.